receiver/chronyreceiver/internal/chrony/client_test.go

```diff
@@ -0,0 +1,196 @@
+		{scenario: "missing protocol", addr: "localhost:323", toError: true},
+		{scenario: "existing socket to potentially connect to", addr: fmt.Sprint("unix://", t.TempDir()), toError: false},
+		{scenario: "invalid path", addr: "unix:///no/socket", toError: true},
+		{scenario: "invalid protocol", addr: "http:/localhost:323", toError: true},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.scenario, func(t *testing.T) {
+			t.Parallel()
+			cl, err := New(tc.addr, time.Second)
+			if tc.toError {
+				assert.Error(t, err, "Must error")
+				assert.Nil(t, cl, "Must have a nil client")
+			} else {
+				assert.NoError(t, err, "Must not error")
+				assert.NotNil(t, cl, "Must have a client")
+			}
+		})
+	}
+}
+
+func TestGettingTrackingData(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		scenario string
+		handler  func(conn net.Conn) error
+		timeout  time.Duration
+		data     *Tracking
+		err      error
+	}{
+		{
+			scenario: "Successful read binary from socket",
+			timeout:  10 * time.Second,
+			handler: func(conn net.Conn) error {
+				type response struct {
+					ReplyHead
+					replyTrackingContent
+				}
+				resp := &response{
+					ReplyHead: ReplyHead{
+						Version: 6,
+						Status:  successfulRequest,
+						Reply:   replyTrackingCode,
+					},
+					replyTrackingContent: replyTrackingContent{
+						RefID: 100,
+						IPAddr: ipAddr{
+							IP:     [16]uint8{127, 0, 0, 1},
+							Family: ipAddrInet4,
+						},
+						Stratum:    10,
+						LeapStatus: 0,
+						RefTime: timeSpec{
+							100, 10, 0,
+						},
+						CurrentCorrection:  binaryFloat(1300),
+						LastOffset:         binaryFloat(10000),
+						RMSOffset:          binaryFloat(12000),
+						FreqPPM:            binaryFloat(3300),
+						ResidFreqPPM:       binaryFloat(123456),
+						SkewPPM:            binaryFloat(9943),
+						RootDelay:          binaryFloat(-1220),
+						RootDispersion:     binaryFloat(-1100000),
+						LastUpdateInterval: binaryFloat(120),
+					},
+				}
+
+				return binary.Write(conn, binary.BigEndian, resp)
+			},
+			data: &Tracking{
+				RefID:              100,
+				IPAddr:             net.IP([]byte{127, 0, 0, 1}),
+				Stratum:            10,
+				LeapStatus:         0,
+				RefTime:            (&timeSpec{100, 10, 0}).Time(),
+				CurrentCorrection:  binaryFloat(1300).Float(),
+				LastOffset:         binaryFloat(10000).Float(),
+				RMSOffset:          binaryFloat(12000).Float(),
+				FreqPPM:            binaryFloat(3300).Float(),
+				ResidFreqPPM:       binaryFloat(123456).Float(),
+				SkewPPM:            binaryFloat(9943).Float(),
+				RootDelay:          binaryFloat(-1220).Float(),
+				RootDispersion:     binaryFloat(-1100000).Float(),
+				LastUpdateInterval: binaryFloat(120).Float(),
+			},
+			err: nil,
+		},
+		{
+			scenario: "Timeout waiting for response",
+			timeout:  time.Second,
+			handler: func(conn net.Conn) error {
+				time.Sleep(2 * time.Second)
```

Comment:
You can use values waaay lower. Like, a timeout of 10ms and a sleep of 50ms. Waiting 2s for nothing isn't going to do well for our already long CI wait times.